File: 2024/Python/Challenge#1.py
from typing import List

def prepare_gifts(gifts: List[int]) -> List[int]:
    non_duplicated_gifts = list(set(gifts))
    sorted_gifts = sorted(non_duplicated_gifts)
    return sorted_gifts


gifts1 = [3, 1, 2, 3, 4, 2, 5]
preparedGifts1 = prepare_gifts(gifts1)
print(preparedGifts1) # [1, 2, 3, 4, 5]

gifts2 = [6, 5, 5, 5, 5]
preparedGifts2 = prepare_gifts(gifts2)
print(preparedGifts2) # [5, 6]

gifts3 = []
preparedGifts3 = prepare_gifts(gifts3)
print(preparedGifts3) # []
# There are no gifts, the list remains empty

File: 2024/TypeScript/Challenge#1.ts

function prepareGifts(gifts: number[]): number[] {
    const nonDuplicatedGifts = [...new Set(gifts)];
    const sortedGifts = [...nonDuplicatedGifts].sort((a, b) => a - b);
    return sortedGifts;
}


const gifts1: number[] = [3, 1, 2, 3, 4, 2, 5];
const preparedGifts1: number[] = prepareGifts(gifts1);
console.log(preparedGifts1); // [1, 2, 3, 4, 5]

const gifts2: number[] = [6, 5, 5, 5, 5];
const preparedGifts2: number[] = prepareGifts(gifts2);
console.log(preparedGifts2); // [5, 6]

const gifts3: number[] = [];
const preparedGifts3: number[] = prepareGifts(gifts3);
console.log(preparedGifts3); // []
// There are no gifts, the list remains empty
